### PR TITLE
Support `scoped` keyword in `using` statements.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #62039.
https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#syntax

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler Please review

**Review by chsienki — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs [hunk 1]
```diff
@@ -20277,6 +20280,804 @@ static RefByteContainer M42(scoped ref ByteContainer bc)
...
+            var source =
+@"#pragma warning disable 219
+
+class Program
+{
+    static void F(ref R r)
+    {
+        using (scoped R r1 = default) {}
+        using (scoped ref R r2 = ref r) {}
+        using (scoped ref readonly R r5 = ref r) {}
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: >using (scoped ref readonly R r5 = ref r) {} [](http://example.com/codeflow?start=8&length=43)

Do we have any binding tests for `using` declarations and statements with `ref` without `scoped`? It looks like parsing succeeds in those cases and we expect binding to report an error but it doesn't look like we have tests that confirm that.
```csharp
static void F(ref R r1)
{
    using ref R r2 = ref r1;
    using ref readonly R r3 = ref r1;
    using (ref R r4 = ref r1) { }
    using (ref readonly R r5 = ref r1) { }
}
```

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: > It looks like parsing succeeds in those cases and we expect binding to report an error but it doesn't look like we have tests that confirm that.

This is how it works, i.e. a binding error is reported. Creating tests for legacy scenarios is out of scope of this PR.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: It is the same error as the one reported with scoped:
```
// (8,23): error CS1073: Unexpected token 'ref
```
And reported by the same existing code.

Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: > It is the same error as the one reported with scoped:

I understand, but I couldn't find any tests that confirmed that.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: > I understand, but I couldn't find any tests that confirmed that.

I am sure there are other ways to confirm this legacy behavior.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:20370]: If the confirmation is what you are after, the behavior can be observed here [https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgNwgCcACUgXhJgFMB3EgJQAoBKAbgFgAoAAQAYSTItQBmjMgCYSlYWKIsSAbwC+3bmiIBXAMYJxIEjwCMAOgCSAESgoADgHsUEYABtq3RdxJfDAZkMAWEitbB2pWJVUuZSA=](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgNwgCcACUgXhJgFMB3EgJQAoBKAbgFgAoAAQAYSTItQBmjMgCYSlYWKIsSAbwC+3bmiIBXAMYJxIEjwCMAOgCSAESgoADgHsUEYABtq3RdxJfDAZkMAWEitbB2pWJVUuZSA=)

**Review by cston — APPROVED** [review]

**Pull request merged**